code/modules/mob/living/basic/boss/thing/thing.dm

```diff
@@ -0,0 +1,268 @@
+#define PHASEREGEN_FILTER "healing_glow"
+#define RUIN_QUEUE "the_thing_depleter"
+/mob/living/basic/boss/thing
+	name = "\improper Thing"
+	icon = 'icons/mob/simple/icemoon/thething.dmi'
+	icon_state = "p1"
+	icon_dead = "dead"
+	gender = NEUTER
+	maxHealth = 1800 //nicely divisible by three
+	health = 1800
+	armour_penetration = 40
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	sharpness = SHARP_EDGED
+	melee_attack_cooldown = CLICK_CD_SLOW
+	attack_verb_continuous = "eviscerates"
+	attack_verb_simple = "eviscerate"
+	attack_sound = 'sound/items/weapons/bladeslice.ogg'
+	attack_vis_effect = ATTACK_EFFECT_SLASH
+	speed = 3.5 //dont make this any faster PLEASE
+	gps_name = "L-4 Biohazard Beacon"
+	ai_controller = /datum/ai_controller/basic_controller/thing_boss
+	loot = list(/obj/item/keycard/thing_boss)
+	crusher_loot = list(/obj/item/keycard/thing_boss, /obj/item/crusher_trophy/flesh_glob)
+	/// Current phase of the boss fight
+	var/phase = 1
+	/// Time the Thing will be invulnerable between phases
+	var/phase_invul_time = 10 SECONDS
+	/// timer of phase invulnerability between phases
+	var/phase_invulnerability_timer
+
+	// ruin logic
+
+	/// if true, this boss may only be killed proper in its ruin by the associated machines as part of the bossfight. Turn off if admin shitspawn
+	var/ruin_spawned = TRUE
```

Comment:
Wouldn't it be better to simply check if it was maploaded or not? It is not like you can turn this off before spawning in anyway.